ts/components/conversation/ImageGrid.tsx

```diff
@@ -1,263 +1,142 @@
 import React, { useContext } from 'react';
-import classNames from 'classnames';
 
 import {
   areAllAttachmentsVisual,
   AttachmentType,
   AttachmentTypeWithPath,
   getAlt,
-  getImageDimensionsInAttachment,
   getThumbnailUrl,
   isVideoAttachment,
 } from '../../types/Attachment';
 
 import { Image } from './Image';
 import { IsMessageVisibleContext } from './message/message-content/MessageContent';
+import styled from 'styled-components';
+import { THUMBNAIL_SIDE } from '../../types/attachments/VisualAttachment';
 
 type Props = {
   attachments: Array<AttachmentTypeWithPath>;
-  bottomOverlay?: boolean;
   onError: () => void;
   onClickAttachment?: (attachment: AttachmentTypeWithPath | AttachmentType) => void;
 };
+
+const StyledImageGrid = styled.div`
+  display: inline-flex;
+  flex-direction: row;
+  align-items: center;
+  gap: var(--margins-sm);
+`;
+
+const StyledImageGridColumn = styled.div`
+  display: inline-flex;
+  flex-direction: column;
+  align-items: center;
+  gap: var(--margins-sm);
+`;
+
 // tslint:disable: cyclomatic-complexity max-func-body-length use-simple-attributes
 export const ImageGrid = (props: Props) => {
-  const { attachments, bottomOverlay, onError, onClickAttachment } = props;
+  const { attachments, onError, onClickAttachment } = props;
 
   const isMessageVisible = useContext(IsMessageVisibleContext);
 
-  const withBottomOverlay = Boolean(bottomOverlay);
-
   if (!attachments || !attachments.length) {
     return null;
   }
 
-  if (attachments.length === 1 || !areAllAttachmentsVisual(attachments)) {
-    const { height, width } = getImageDimensionsInAttachment(attachments[0]);
+  const shared = {
+    onClick: onClickAttachment,
+    onError: onError,
+    softCorners: true,
+  };
 
+  if (attachments.length === 1 || !areAllAttachmentsVisual(attachments)) {
     return (
-      <div className={classNames('module-image-grid', 'module-image-grid--one-image')}>
+      <StyledImageGrid>
         <Image
           alt={getAlt(attachments[0])}
-          bottomOverlay={withBottomOverlay}
           attachment={attachments[0]}
           playIconOverlay={isVideoAttachment(attachments[0])}
-          height={height}
-          width={width}
+          height={THUMBNAIL_SIDE}
+          width={THUMBNAIL_SIDE}
           url={isMessageVisible ? getThumbnailUrl(attachments[0]) : undefined}
-          onClick={onClickAttachment}
-          onError={onError}
           attachmentIndex={0}
+          {...shared}
         />
-      </div>
+      </StyledImageGrid>
     );
   }
 
   if (attachments.length === 2) {
+    // when we got 2 attachments we render them side by side with the full size of THUMBNAIL_SIDE
     return (
-      <div className="module-image-grid">
+      <StyledImageGrid>
         <Image
```

Comment:
We should probably use map() on attachments to render the 2 attachments instead of duplicating the components. Also don't forget to add the key prop if you do use map

Reply:
:+1: